packages/amplify_datastore/example/integration_test/separate_integration_tests/basic_model_operation_test.dart

```diff
@@ -0,0 +1,128 @@
+
+      if (enableCloudSync) {
+        // set an async getter to retrieve a desired hub event with a speicfic
+        // event matcher
+        var eventGetter = getExpectedSubscriptionDataProcessedEvent(
+          eventMatcher: (event) =>
+              (event.element.model as Blog).id == testBlog.id &&
+              // newly saved model at this step should have version: 1
+              event.element.version == 1,
+        );
+
+        // save model locally and to sync to cloud
+        await Amplify.DataStore.save(testBlog);
+
+        // wait for the desired event to arrive
+        var event = await eventGetter;
+        expect(event.element.deleted, isFalse);
+      } else {
+        await Amplify.DataStore.save(testBlog);
+      }
+
+      var blogs = await Amplify.DataStore.query(Blog.classType);
+      expect(blogs.length, 1);
+      expect(blogs.contains(testBlog), isTrue);
+    });
+
+    testWidgets(
+      'should update existing model ${enableCloudSync ? 'and sync to cloud' : ''}',
+      (WidgetTester tester) async {
+        // get previously saved model
+        var testBlog = (await Amplify.DataStore.query(Blog.classType))[0];
+        // update model
+        var updatedTestBlog = testBlog.copyWith(name: "updated test blog");
+
+        if (enableCloudSync) {
+          var eventGetter = getExpectedSubscriptionDataProcessedEvent(
+            eventMatcher: (event) =>
+                (event.element.model as Blog).id == updatedTestBlog.id &&
+                // updated model at this step should have version: 2
+                event.element.version == 2,
+          );
+
+          await Amplify.DataStore.save(updatedTestBlog);
+
+          var event = await eventGetter;
+          expect(event.element.deleted, isFalse);
+        } else {
+          await Amplify.DataStore.save(updatedTestBlog);
+        }
+
+        var updatedBlogs = await Amplify.DataStore.query(Blog.classType);
+
+        // verify blog was updated
+        expect(updatedBlogs.length, 1);
+        expect(updatedBlogs.contains(updatedTestBlog), isTrue);
+      },
+    );
+
+    testWidgets(
+      'should delete existing model ${enableCloudSync ? 'and sync to cloud' : ''}',
+      (WidgetTester tester) async {
+        // get previously saved model
+        var testBlog = (await Amplify.DataStore.query(Blog.classType))[0];
+
+        if (enableCloudSync) {
+          var eventGetter = getExpectedSubscriptionDataProcessedEvent(
+            eventMatcher: (event) =>
+                (event.element.model as Blog).id == testBlog.id &&
+                // deleted model at this step should have version: 3
+                event.element.version == 3,
+          );
+
+          await Amplify.DataStore.delete(testBlog);
+
+          var event = await eventGetter;
+          expect(event.element.deleted, isTrue);
+        } else {
+          await Amplify.DataStore.delete(testBlog);
+        }
+
+        var blogs = await Amplify.DataStore.query(Blog.classType);
```

Comment:
question: Would it be better to try to query for the one you deleted in case race condition makes the backend not empty?